File: Leetcode Questions/Medium/11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int area=0;
        int l=0,h=height.size()-1;
        while(l<h){
            if(height[l]<height[h]){
                int checkArea=height[l]*(h-l);
                if(checkArea>area) area=checkArea;
                l++;
            }
            else {
                int checkArea=height[h]*(h-l);
                if(checkArea>area) area=checkArea; 
                h--;
            }
        }
        return area;
    }
};

File: Leetcode/Medium/PYTHON/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        e=len(height)-1
        area=0
        while(l<e):
            currArea=(e-l)*min(height[e],height[l])
            area=max(area,currArea)
            if(height[e]<height[l]): e-=1
            else: l+=1
        return area
    
#Link: https://leetcode.com/problems/container-with-most-water/